src/auth.js

```diff
@@ -88,8 +96,14 @@ function generateJwt(privateKey, keyPassword, ttl) {
         repository: process.env.GITHUB_REPOSITORY,
         ref: process.env.GITHUB_REF
     };
-    const decryptedKey = rsasign.KEYUTIL.getKey(privateKey, keyPassword);
-    return rsasign.KJUR.jws.JWS.sign(alg, JSON.stringify(header), JSON.stringify(payload), decryptedKey);
+    try{
+        const decryptedKey = rsasign.KEYUTIL.getKey(privateKey, keyPassword);
+        return rsasign.KJUR.jws.JWS.sign(alg, JSON.stringify(header), JSON.stringify(payload), decryptedKey);
+    } catch (err) {
+        const new_error = new Error(`Error generating the jwt., message: ${err?.message}`);
+        new_error.stack += err.stack;
+        throw new_error;
+    }
```

Comment:
Do we know this fails? This is a local operation afaik that doesn't await so I'm not sure how it relates to the promises problem

Reply:
I'll remove it. We don't call it, and I don't remember why I added it